consensus/types/round_state.go

```diff
@@ -116,6 +117,7 @@ func (rs *RoundState) RoundStateSimple() RoundStateSimple {
 		LockedBlockHash:   rs.LockedBlock.Hash(),
 		ValidBlockHash:    rs.ValidBlock.Hash(),
 		Votes:             votesJSON,
+		Proposer:          rs.NewRoundEvent().Proposer,
```

Comment:
I am not sure it's appropriate to use this method here since it's used for something different.

Couldn't we just copy 

```
	addr := rs.Validators.GetProposer().Address
	idx, _ := rs.Validators.GetByAddress(addr)
```

and use them to construct 

```
types.ValidatorInfo{
			Address: addr,
			Index:   idx,
		}
```

Reply:
Sounds good @melekes